verifier/src/lib.rs

```diff
@@ -79,11 +84,38 @@ pub enum VerificationError {
     },
 }
 
+impl DisplayableError for VerificationError {}
+
+#[derive(Debug, Clone)]
+/// A [`CtOption`] wrapped to implement the [`Display`] trait.
+pub struct DisplayableCtOption<T>(CtOption<T>);
+
+impl<T> From<CtOption<T>> for DisplayableCtOption<T> {
+    fn from(ct_option: CtOption<T>) -> Self {
+        Self(ct_option)
+    }
+}
+
+impl<T: DisplayableError> Display for DisplayableCtOption<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        let option: Option<T> = self.0.clone().into();
+        match option {
+            Some(value) => match f.alternate() {
+                true => write!(f, "{value:#}")?,
+                false => write!(f, "{value}")?,
+            },
+            None => write!(f, "passed")?,
```

Comment:
Chose to not dig into passing options, figured `Debug` could be used for that, even though one would have to look at 0 vs 1 for the ct option.